lib/request.js

```diff
@@ -769,7 +769,9 @@ AWS.Request.addPromisesToClass = function addPromisesToClass(PromiseDependency)
         if (resp.error) {
           reject(resp.error);
         } else {
-          resolve(resp.data);
+          var data = AWS.util.copy(resp.data);
```

Comment:
This has messed up our mocking of the AWS SDK (which relies on... some messiness):

https://gist.github.com/wryun/347283d8e715afc4f904ffe5838c536c

Do you have any suggestions for how to do this in a nicer way?